### PR TITLE
fix: [N04] Complete unfinished tests

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

OZ identified the following issue:

```
We identified two unfinished tests that handle relevant edge cases.
```

This PR addresses this issue by adding "Correctly sends bond to store if voted down" test.

Note that "Correctly handles rolled votes" test has already been added in #4329.

### CODE REVIEW

File: packages/core/test/hardhat/data-verification-mechanism/SpamGuard.js [hunk 1]
```diff
@@ -483,6 +486,62 @@ describe("SpamGuard", function () {
     assert.equal(await votingToken.methods.balanceOf(account2).call(), toWei("0")); // 0mm.
   });
 
-  it("Correctly sends bond to store if voted down", async function () {});
```
Comment (chrismaree) [inline packages/core/test/hardhat/data-verification-mechanism/SpamGuard.js:486]: two comments:
1) this can be simplified. I don't think you need a) two requests or b) so many votes. if you want to keep both and all the votes then at least check that the price settles on the one that is not deleted.
2) this is the negative test: check that the store gets the bond if voted down. we should also check that the proposer gets it back if voted up in a separate test.

Other than this, looks good!

Comment (Reinis-FRP) [reply packages/core/test/hardhat/data-verification-mechanism/SpamGuard.js:486]: > this can be simplified. I don't think you need a) two requests or b) so many votes. if you want to keep both and all the votes then at least check that the price settles on the one that is not deleted.

This still requires 2 requests since upper bound in spam deletion range cannot be 0 (`signalRequestsAsSpamForDeletion([[0,0]])` would revert).

But agree that this test does not require actually voting on other requests than the spam deletion proposal

Comment (Reinis-FRP) [reply packages/core/test/hardhat/data-verification-mechanism/SpamGuard.js:486]: > this is the negative test: check that the store gets the bond if voted down. we should also check that the proposer gets it back if voted up in a separate test.

Isn't it already covered in `Correct bonds refund` test above?

Comment (Reinis-FRP) [reply packages/core/test/hardhat/data-verification-mechanism/SpamGuard.js:486]: Removed voting on price requests in 0c571c047bca388eebc1448f5e4c517658e9eb52

**Comment by chrismaree** [comment]
closed as this PR is now redundant due to #4352
